pkg/exporter/process.go

```diff
@@ -282,25 +284,36 @@ func (ep *ExportingProcess) createAndSendMsg(set entities.Set) (int, error) {
 	}
 	msg.SetSequenceNum(ep.seqNumber)
 
-	bytesSlice := make([]byte, msgLen)
-	copy(bytesSlice[:entities.MsgHeaderLength], msg.GetMsgHeader())
-	copy(bytesSlice[entities.MsgHeaderLength:entities.MsgHeaderLength+entities.SetHeaderLen], set.GetHeaderBuffer())
-	index := entities.MsgHeaderLength + entities.SetHeaderLen
-	for _, record := range set.GetRecords() {
-		len := record.GetRecordLength()
-		copy(bytesSlice[index:index+len], record.GetBuffer())
-		index += len
+	//bytesSlice := make([]byte, msgLen)
+	//copy(bytesSlice[:entities.MsgHeaderLength], msg.GetMsgHeader())
+	//copy(bytesSlice[entities.MsgHeaderLength:entities.MsgHeaderLength+entities.SetHeaderLen], set.GetHeaderBuffer())
+	//index := entities.MsgHeaderLength + entities.SetHeaderLen
+	//for _, record := range set.GetRecords() {
+	//	len := record.GetRecordLength()
+	//	copy(bytesSlice[index:index+len], record.GetBuffer())
+	//	index += len
+	//}
+
+	message := make(map[string]interface{})
+	record := set.GetRecords()[0]
+	for _, element := range record.GetOrderedElementList() {
+		message[element.Element.Name] = element.Value
 	}
+	message["@timestamp"] = time.Now().Format(time.RFC3339)
+	writer := bytes.NewBuffer(make([]byte, 0, 1.2*300 * msgLen))
```

Comment:
we don't need to have such capacity. We could directly specify msgLen like below:

`writer := bytes.NewBuffer(make([]byte, msgLen))`

I know it is a draft. Just pitching in..